backend/game/engine/board.py

```diff
@@ -29,6 +30,8 @@ def turn(self, active_player):
         if active_player != self.players.get_current_player():
             raise Exception("It is not %s's turn" % (id(active_player)))
         self.turn_actions(active_player)
+        if active_player.has_instance_of_card(SolarPowered()):
```

Comment:
Although this is where the card would be active, I think writing a separate function for all end of turn cards would be best for now as I can see this part of the code expanding or us taking it in a different direction so would want it to be slightly separated from the main turn function

Reply:
I was just thinking about that. I think that should be a future refactor since these requests have just lived way too long and this can be good enough for now.